tests/python/unittest/test_sparse_operator.py

```diff
@@ -1217,6 +1218,13 @@ def test_dot_csr(lhs_shape, rhs_shape, rhs_stype, trans_lhs, lhs_density, rhs_de
             test_dot_csr(lhs_shape, (lhs_shape[0], rnd.randint(1, 10)), 'row_sparse', True, lhs_d, rhs_d)
 
 
+def test_sparse_dot_twice():
```

Comment:
Hmm. I'm curious why the existing unit tests don't detect this bug. 
Also shall we write a test explicit test the root cause we found, instead of seeding numpy? Seeding is not reliable in my opinion - if someone else changed `test_sparse_dot ` with one more np random array created in the middle of the test, it might not still be testing what we wanted it to test on.

Reply:
I agree here.  I recommend creating some input that creates a 0 nnr, then proving the test fails on the existing codebase, then finally is fixed by the PR.  I supplied the seeded test case just as a way to reproduce the problem, not as a reliable way to test the function going forward.